src/FsCheck/Common.fs

```diff
@@ -25,12 +30,22 @@ module internal Common =
                         let res = f n
                         memo.Add(n,res)
                         res)
+#else
+    open System.Collections.Concurrent
+
+    ///Create the Memoizer
+    let memoizer () = new ConcurrentDictionary<_,_>()
+
+    ///Memoize the given function using the given dictionary
+    let memoizeWith (memo:ConcurrentDictionary<'a,'b>) (f: 'a -> 'b) =
```

Comment:
ConcurrentDictionary only uses locks on inserts IIRC. A minor improvement for FsCheck as there should be little contention in test code. Still nice IMO